src/app/tests/api/v1/test_codegen.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+from httpx import AsyncClient
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.core.config import settings
+from app.tests.utils.models import create_project, create_generation_template, create_welding_point
+
+pytestmark = pytest.mark.asyncio
+
+
+async def test_codegen(client: AsyncClient, database: AsyncSession):
+    project_obj = await create_project(db=database)
+    project_id = project_obj.id
+
+    welding_point_obj = await create_welding_point(db=database, project_obj=project_obj)
+    generation_template_obj = await create_generation_template(db=database)
+    generation_template_id = generation_template_obj.id
+
+    data = {"generation_template_id": project_id, "project_id": generation_template_id}
+    response = await client.post(f"{settings.API_V1_STR}/codegeneration/generate", json=data)
+    assert response.status_code == 200
+
+    content = response.json()
+    assert content == f"{welding_point_obj.x}, {welding_point_obj.y}, {welding_point_obj.z} / " \
+                      f"{welding_point_obj.roll}, {welding_point_obj.pitch}, {welding_point_obj.yaw} / " \
+                      f"{welding_point_obj.welding_order}"
```

Comment:
Yes and no. `unittest` and `pytest` are two distinct testing frameworks. It is basically possible to use unittest with pytest, but according to my research it is not possible to use fixtures in class methods (class methods would be the tests) directly. We would have to make a special fixture within the class to get the fixtures or mark the whole class to get the fixtures. But then the database and client would be used for the whole class, what I don't want to do. I like that we get for each test a whole new database. It would be possible, but we would need more boilerplate, which I personally would try to avoid except when we really gain something from. 

In my opinion, the current error output is sufficient for our case and I do not see the use in us to write more boilerplate for a little more info. For real applications it would make sense to look again at this topic :D

We could also print out e.g. the response every time as a trade-off. The print output is only displayed on error.

Reply:
Okay if it's not feasible it's not feasible :D